docs/src/pages/system/sx/sx.md

```diff
@@ -0,0 +1,92 @@
+# The `sx` prop
+
+<p class="description">Utility for styling elements inline, using the theme values.</p>
+
+## Getting Started
+
+Each `@material-ui/core` component supports the `sx` prop. The prop is a superset of CSS, that let's you add any CSS to the underlaying component, while mapping values to different theme keys. This should help you to keep your application look consistent.
```

Comment:
I'm not sure I would start with mentioning `@material-ui/core`.

Reply:
What do you think of this:

> The `sx` prop is a superset of CSS, that let's you add any CSS to the underlaying component, while mapping values to different theme keys. This should help you to keep your application look consistent. The prop is available in all `@material-ui/core` components. 


Later when we support this on all primitives, we can add that here too. Or at least, we can export the `styleFunctionSx` and add something like:

> You can include this prop on your own components too by using the `styleFunctionSx` style function from `@material-ui/core/Box`, or you can automatically add it to your component by using the `experimentalStyled` from `@material-ui/core/styles`.
> ```
> import { styleFunctionSx } from '@material-ui/core/Box';
> import styled from '@emotion/styled';
> 
> const Div = styled('div')(styleFunctionSx);
> ```
> or
> ```
> import { experimentalStyled as styled } from '@material-ui/core/styles';
> 
> const Div = styled('div')``;
> ```

Reply:
I'l wait for it to be updated before correcting this.

Reply:
We haven't decided on adding `sx` to every component nor have we released it. We shouldn't document something before.

Reply:
> We haven't decided on adding sx to every component nor have we released it. We shouldn't document something before.

The Slider is meant to be a projection of how all the other components will be architectured. Slider now supports the sx prop and was released. So I think that in theory, this decision and release event has already happened. I would argue that the decision started 2 years ago with

> In the future, we will look into using it directly in the core components.

https://medium.com/material-ui/introducing-material-ui-design-system-93e921beb8df

"look into" as how can we make it fast enough. Then about a year ago with (#15561).

Having a look at all the issues that where closed linking to this issue is interesting to gauge the pain.

Reply:
> So I think that in theory, this decision and release event has already happened. 

That is not how an alpha works. Its goal is to give us freedom to add and remove things without having to release a major version every time. If you're now flipped to "everything released in alpha is stable" then you should've told us that before we started working on v5. That's a pretty fundamental change to the alpha process especially if it involves decisions we haven't discussed.

As ususal: something existing is in no way shape or form under any circumstance and argument for its existence. Slider with sx is not a good API just because it shipped with it. The original PR still doesn't include a real world use case and just some toying around.

> I would argue that the decision started 2 years ago with

A decision isn't a process. Could you link me the conclusion of the discussion with having widgets with system props that just affect the root container?

Reply:
> Its goal is to give us freedom to add and remove things without having to release a major version every time.

Agree, the objective of pre-releases is to allow a short feedback loop. It allows us to validate ideas directly with developers. On this one, we are waiting to get insight from developers after having released it. It seems that the trend of improvement is toward supporting `sx` everywhere: #23220. The value of the prop increases the more it can be used in a systematic manner.

---

I don't understand the concern with the `sx` prop, the value proposition started to be discussed in #22819, then #23053, e.g. https://github.com/mui-org/material-ui/pull/23053#issuecomment-710095357.

Reply:
> I don't understand the concern with the sx prop

It's JavaScript. JavaScript is only good if it solves a problem. On the flipside: I do not understand how you don't see a problem with adding features after maintaining open-source for 3+ years.